manager/manager/typeresolver.go

```diff
@@ -92,18 +93,27 @@ func (tr *typeResolver) ResolveTypes(config *common.Configuration, imports []*co
 	}
```

Comment:
Ok, take a look. Pulled out template generation code. Other than that it's probably going to be a major refactor.